### ISSUE TITLE
Latency Loss in SupernetLoss

### ISSUE THREAD

**Issue by latifisalar** [AnnaAraslanova/FBNet#3]
There is an inconsistency in implementation of latency loss compared to original paper:
 In `forward` function of `SupernetLoss`, shouldn't we have `lat = torch.log(latency) ** self.beta` instead of `lat = torch.log(latency ** self.beta)`? I made the changes as so, but latency loss kept increasing during the search.

**Comment by Catosine** [comment]
@latifisalar Hi! Here is the thing: torch.log is the function for **logarithm**. According to some math knowledge, log(a^b) is actually b*log(a) but not log(a)^b.

GL,

**Comment by latifisalar** [comment]
@Catosine Your statement is correct, but I don't understand how is that related to my question! Based on the Eq. 2 of paper, loss function uses log(lat)^beta, whereas in this repository, it is implemented as log(lat^beta).

**Comment by Catosine** [comment]
@latifisalar Sorry man, it's my bad.
In general, I think this implementation is different from the paper, or even just wrong at some certain point. The validation accuracy on Cifar10 also proves that.
GL,

**Comment by latifisalar** [comment]
The problem is that it is using the backbone and hyper-parameters of ImageNet for Cifar10. For example, the backbone has 5 convs with stride of 2, so at the end it ends up with a spatial size of 1, compared to ImageNet which gets 7.
I changed the backbone to MobileNetV2 and by changing the hyper-parameters, I can get ~95% accuracy.

**Comment by AnnaAraslanova** [comment]
@latifisalar You are right! It's a huge misprint =( I will mention it on the main page

**Comment by iamweiweishi** [comment]
> The problem is that it is using the backbone and hyper-parameters of ImageNet for Cifar10. For example, the backbone has 5 convs with stride of 2, so at the end it ends up with a spatial size of 1, compared to ImageNet which gets 7.
> I changed the backbone to MobileNetV2 and by changing the hyper-parameters, I can get ~95% accuracy.

@latifisalar Hi, I followed your suggestion and tried to modify the backbone for Cifar10, but the accuracy of VALIDATION still stays below 90%, could pls give me some tip on how to adjust the hyper-parameters here? 
Many thanks.

**Comment by latifisalar** [comment]
Here is the setup that I'm using:

Backbone:
- First stage ConvBNRelu: [channels: 32, stride: 1]

- Search stages: 

        ("channel_size", [16,
                      24, 24,
                      32, 32, 32,
                      64, 64, 64, 64,
                      96, 96, 96,
                      160, 160, 160,
                      320]),
        ("strides", [1,
                     1, 1,
                     2, 1, 1,
                     2, 1, 1, 1,
                     1, 1, 1,
                     2, 1, 1,
                     1])

- Last stage conv2d: [channels: 1280, stride: 1]

architecture_functions/config_for_arch.py:

-     'optimizer' : {
        'lr' : 0.1,
        'momentum' : 0.9,
        'weight_decay' : 1 * 1e-4
        },

-     'train_settings' : {
       'scheduler' : 'CosineAnnealingLR',
       'cnt_epochs' : 600,
       'eta_min' : 0.0001,
       ...
       },

I think these are all the changes that I have made.

**Comment by weihui98** [comment]
> Here is the setup that I'm using:
> 
> Backbone:
> 
>     * First stage ConvBNRelu: [channels: 32, stride: 1]
> 
>     * Search stages:
>       ```
>         ("channel_size", [16,
>                       24, 24,
>                       32, 32, 32,
>                       64, 64, 64, 64,
>                       96, 96, 96,
>                       160, 160, 160,
>                       320]),
>         ("strides", [1,
>                      1, 1,
>                      2, 1, 1,
>                      2, 1, 1, 1,
>                      1, 1, 1,
>                      2, 1, 1,
>                      1])
>       ```
> 
>     * Last stage conv2d: [channels: 1280, stride: 1]
> 
> 
> architecture_functions/config_for_arch.py:
> 
>     * ```
>       'optimizer' : {
>         'lr' : 0.1,
>         'momentum' : 0.9,
>         'weight_decay' : 1 * 1e-4
>         },
>       ```
> 
>     * ```
>       'train_settings' : {
>        'scheduler' : 'CosineAnnealingLR',
>        'cnt_epochs' : 600,
>        'eta_min' : 0.0001,
>        ...
>        },
>       ```
> 
> 
> I think these are all the changes that I have made.

Hello! Could you please give me some more details about CIFAR-10? For example, "input_shape", "channel_size" and "strides" , counterpartly. 

Thank you very much!!
